src/components/Todo/common/Todo/Item/index.tsx

```diff
@@ -88,7 +90,11 @@ export const Item: FC<Props> = ({
         className={classes.reorderItem}
         dragListener={draggable}
         onDragEnd={() => setDraggable(false)}
-        style={{ boxShadow, y }}
+        style={{ boxShadow, y , left: xPosition? 30: 0}}
```

Comment:
thanks @Curis-lab for working on this.
I tried it and the drag and drop implementation looks good 
We need to handle few more cases
1. The X button should not move 
2. Hierarchy support means the data strucuture storing data and the callback on data change should also represent the hierarchy. This means you should have a way to persist the hierarchy information also.
This mean both default iterms data structure and callback should change.
`<Todo defaultItems={[]} onChange={mockHandleChange} />`